src/main/java/org/jenkinsci/plugins/codesonar/services/AlertsService.java

```diff
@@ -0,0 +1,110 @@
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.apache.http.client.utils.URIBuilder;
+import org.jenkinsci.plugins.codesonar.CodeSonarPluginException;
+import org.jenkinsci.plugins.codesonar.ResponseErrorException;
+import org.jenkinsci.plugins.codesonar.api.CodeSonarAlertCounter;
+import org.jenkinsci.plugins.codesonar.models.json.CodeSonarAlertData;
+
+import com.google.common.base.Throwables;
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+
+public class AlertsService {
+    private static final Logger LOGGER = Logger.getLogger(AlertsService.class.getName());
+    
+    final private HttpService httpService;
+
+    public AlertsService(HttpService httpService) {
+        this.httpService = httpService;
+    }
+    
+    public CodeSonarAlertCounter getAlertCounter(URI baseHubUri, long analysisId) throws IOException {
+        CodeSonarAlertCounter counter = new CodeSonarAlertCounter();
+        //Loop through all possible kind values, currently from 0 to 26. Upper bound 50 is for future alert kinds.
+        for(long kind = 0; kind < 50; kind++) {
+            //Build char_table request URL
+            URIBuilder uriBuilder = new URIBuilder(baseHubUri);
+            uriBuilder.setPath(String.format("/analysis/%d-alert%d.json", analysisId, kind));
+            
+            URI requestUri = null;
+            try {
+                requestUri = uriBuilder.build();
+            } catch (URISyntaxException e) {
+                LOGGER.log(Level.WARNING, "Request URI for Alerts API contains a syntax error. %nException: {0}%nStack Trace: {1}", new Object[] {e.getMessage(), Throwables.getStackTraceAsString(e)});
+                return null;
+            }
+            
+            LOGGER.log(Level.INFO, "Querying alerts with kind={0}", kind);
+            InputStream jsonContent = null;
+            try {
+                jsonContent = httpService.getContentFromUrlAsInputStream(requestUri.toASCIIString());
+            } catch(ResponseErrorException e) {
+                if(e.getStatus() == 404) {
+                    /*
+                     * A 404 response is returned anytime the request specifies an alert kind that has no
+                     * occurrences on the specified analysis
+                     */
+                    LOGGER.log(Level.INFO, "Skipping current alert kind", e);
+                    continue;
+                } else if(e.getStatus() == 500) {
+                    /*
+                     * A 500 response is returned anytime the request specifies an alert kind that is out
+                     * of range with respect to all currently supported alert kinds.
+                     * This makes status 500 the right one to establish when to stop looping through kinds.
+                     */
+                    LOGGER.log(Level.INFO, MessageFormat.format("Stop looping through alert kinds at kind id {0}", kind), e);
+                    break;
```

Comment:
I think 500 is a generic error.  We might get this error for reasons other than the kind number being out of range.  I think we will need to look at the response body to determine the cause of the error.  No matter whether we can or cannot use the response body to automatically determine the nature of the error, we should always print it to the log.